### ISSUE TITLE
Not unambiguous selection of a line with peptide

### ISSUE THREAD

**Issue by serge2016** [griffithlab/pVACtools#361]
Hello!
I was trying to get the results according to https://pvactools.readthedocs.io/en/latest/pvacseq/getting_started.html.
I can't get the same results as are in file `tools/pvacseq/example_data/MHC_Class_I/Test.filtered.condensed.ranked.tsv`. There are 2 differences:
1. The format of values in columns `MT IC50` and `WT IC50` now looks to be `.3f` and the file in the repo there are many digits after the point.
2. If we sort the table in `tools/pvacseq/example_data/MHC_Class_I/Test.all_epitopes.tsv` by `Median MT Score` then we'll see that there are lines with identical value in this column, including the first pair of lines (after sorting). It's not unambiguous, how one of them is selected in this code:
https://github.com/griffithlab/pVACtools/blob/c9a76aaebc10f817c366f2f1b61c6f03a3ac0fa0/lib/top_score_filter.py#L28
I suppose that it would be better to add sorting e.g. by `HGVSc` before selecting the line only by `Median MT Score` value.

**Comment by susannasiebert** [comment]
1.  I must've failed to update the example data for the latest releases. I will make sure to update it for the next one.
2. Between binding filtering and top-score filtering we also apply VAF and expression filtering. If a variant has multiple epitopes left after those filters then the epitope with the lowest IC50 value will be selected by the top score filter. If multiple epitopes with the same IC50 for a variant remain (e.g., in cases where you have multiple transcripts that result in the same epitope prediction), we select the first one that is encountered. A variant is identified by the top score filter as having the same chr- start - stop - ref - alt (see the code just a few lines above the line you posted).

I'm not exactly sure what you are suggesting with sorting by HGVSc. Please elaborate.

**Comment by serge2016** [comment]
Take a look at these two lines from file `tools/pvacseq/example_data/MHC_Class_I/Test.all_epitopes.tsv` (I hope it is possible to copy them from here with TABs and to paste into Excel or something like it):
```
Chromosome	Start	Stop	Reference	Variant	Transcript	Transcript Support Level	Ensembl Gene ID	Variant Type	Mutation	Protein Position	Gene Name	HGVSc	HGVSp	HLA Allele	Peptide Length	Sub-peptide Position	Mutation Position	MT Epitope Seq	WT Epitope Seq	Best MT Score Method	Best MT Score	Corresponding WT Score	Corresponding Fold Change	Tumor DNA Depth	Tumor DNA VAF	Tumor RNA Depth	Tumor RNA VAF	Normal Depth	Normal VAF	Gene Expression	Transcript Expression	Median MT Score	Median WT Score	Median Fold Change	MHCflurry WT Score	MHCflurry MT Score	MHCnuggetsI WT Score	MHCnuggetsI MT Score	NetMHC WT Score	NetMHC MT Score	PickPocket WT Score	PickPocket MT Score	SMM WT Score	SMM MT Score	SMMPMBEC WT Score	SMMPMBEC MT Score
22	50615580	50615581	C	T	ENST00000159647.5	NA	ENSG00000073150	missense	S/F	147	PANX2	ENST00000159647.5:c.440C>T	ENSP00000159647.5:p.Ser147Phe	HLA-A*02:01	10	2	9	ALGWEFLAFT	ALGWEFLAST	NetMHC	29,28	42,81	1,462	151	1.0	NA	NA	NA	NA	6.414	NA	160.322	196.834	1.228	897.580843743621	505.701535147578	186.263299120293	166.891875035357	42.81	29.28	503.393672722	367.821932979	136.924133105	102.206849152	207.405371596	153.751726065
22	50615580	50615581	C	T	ENST00000395842.2	NA	ENSG00000073150	missense	S/F	147	PANX2	ENST00000395842.2:c.440C>T	ENSP00000379183.2:p.Ser147Phe	HLA-A*02:01	10	2	9	ALGWEFLAFT	ALGWEFLAST	NetMHC	29,28	42,81	1,462	151	1.0	NA	NA	NA	NA	6.414	NA	160.322	196.834	1.228	897.580843743621	505.701535147578	186.263299120293	166.891875035357	42.81	29.28	503.393672722	367.821932979	136.924133105	102.206849152	207.405371596	153.751726065
```
If this is not the final file before selecting best epitopes, then point to the file, please, that is previous to ranking & condencing.

**Comment by susannasiebert** [comment]
There are four filtering steps (in that order): 
1) binding score
2) coverage + expression
3) transcript support level
4) top-score
We don't output the intermediate file for each filtering step so there isn't a file to point to that has filters 1-3 but not the top score filter. You're welcome to run the filters manually with the commands provided here: https://pvactools.readthedocs.io/en/latest/pvacseq/filter_commands.html, to see which epitopes get filtered out in each step.

For your example the top score level would probably pick the first one since there is no TSL or transcript expression annotation that would eliminate one of the transcripts.

**Comment by serge2016** [comment]
That is exactly my question: what sorting order is used in this particular situation to pick "the first one"? And I think it would be nice to add predefined sorting to such a case by any column, for example by column `HGVSc`.

**Comment by susannasiebert** [comment]
We use Gene Name - Mutation.

**Comment by susannasiebert** [comment]
You do make a good point that it would be better if we resorted in the top score filter to ensure this behavior since the original sorting happens when we create the all_epitopes file. We use pandas query to filter and I can't find documentation that states whether or not it preserves order.

**Comment by susannasiebert** [comment]
Please see issue #362 regarding the progress of adding a sorting step before running the top score filter. PR #365 is currently under review and will be available with the next feature release (1.4.0).

I'm closing this issue in favor of #362
